hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/HazelcastTimestamper.java

```diff
@@ -29,13 +31,27 @@
 public final class HazelcastTimestamper {
 
     private static final int SEC_TO_MS = 1000;
+    private static final int BIN_DIGITS = 12;
+    private static final int ONE_MS = 1 << BIN_DIGITS;
+    private static final AtomicLong VALUE  = new AtomicLong();
 
     private HazelcastTimestamper() {
     }
 
     public static long nextTimestamp(HazelcastInstance instance) {
```

Comment:
Can you add a documentation over the method describing the algorithm used here, what it does, what it guarantees, why it's not just a regular timestamp like `System.currentTimeMillis()`, what `BIN_DIGITS` means, what's stored in `VALUE` .. etc?